pkg/swagger/ginSwagger.go

```diff
@@ -0,0 +1,210 @@
+		c.OAuth = config
+	}
+}
+
+func newConfig(configFns ...func(*Config)) *Config {
+	config := Config{
+		URLs:                 []string{"doc.json", "doc.yaml"},
+		DocExpansion:         "list",
+		DomID:                "swagger-ui",
+		InstanceName:         "swagger",
+		DeepLinking:          true,
+		PersistAuthorization: false,
+		SyntaxHighlight:      true,
+	}
+	for _, fn := range configFns {
+		fn(&config)
+	}
+	if config.InstanceName == "" {
+		config.InstanceName = swag.Name
+	}
+	return &config
+}
+
+var WrapHandler = GinWrapHandler()
+
+func GinWrapHandler(options ...func(*Config)) gin.HandlerFunc {
+	config := newConfig(options...)
+	index, _ := template.New("swagger_index.html").Parse(indexTemplate)
+	re := regexp.MustCompile(`^(.*/)([^?].*)?[?|.]*$`)
+
+	return func(c *gin.Context) {
+		// Set security headers to protect against ClickJacking and XSS
+		c.Header("X-Frame-Options", "DENY")
+		c.Header("X-XSS-Protection", "1; mode=block")
+		c.Header("Content-Security-Policy", "default-src 'self'; script-src 'self' 'unsafe-inline' https://cdn.jsdelivr.net")
+		c.Header("X-Content-Type-Options", "nosniff")
+
+		if c.Request.Method != http.MethodGet {
+			c.AbortWithStatus(http.StatusMethodNotAllowed)
+			return
+		}
+		matches := re.FindStringSubmatch(c.Request.RequestURI)
+		path := matches[2]
+
+		switch filepath.Ext(path) {
+		case ".html":
+			c.Header("Content-Type", "text/html; charset=utf-8")
+		case ".css":
+			c.Header("Content-Type", "text/css; charset=utf-8")
+		case ".js":
+			c.Header("Content-Type", "application/javascript")
+		case ".json":
+			c.Header("Content-Type", "application/json; charset=utf-8")
+		case ".yaml":
+			c.Header("Content-Type", "text/plain; charset=utf-8")
+		case ".png":
+			c.Header("Content-Type", "image/png")
+		}
+
+		response := c.Writer
+		if _, ok := response.(http.Flusher); ok {
+			defer response.(http.Flusher).Flush()
+		}
+
+		switch path {
+		case "index.html":
+			_ = index.Execute(c.Writer, config)
+		case "doc.json":
+			doc, err := swag.ReadDoc(config.InstanceName)
+			if err != nil {
+				c.AbortWithError(http.StatusInternalServerError, err) //nolint:errcheck
+				return
+			}
+			c.String(http.StatusOK, doc)
+		case "doc.yaml":
+			jsonString, err := swag.ReadDoc(config.InstanceName)
+			if err != nil {
+				c.AbortWithError(http.StatusInternalServerError, err) //nolint:errcheck
+				return
+			}
+			doc, err := yaml.JSONToYAML([]byte(jsonString))
+			if err != nil {
+				c.AbortWithError(http.StatusInternalServerError, err) //nolint:errcheck
+				return
+			}
+			c.String(http.StatusOK, string(doc))
+		default:
+			c.Request.URL.Path = matches[2]
+			http.FileServer(http.FS(swaggerFiles.FS)).ServeHTTP(c.Writer, c.Request)
+		}
+	}
+}
+
+const indexTemplate = `<!-- HTML for static distribution bundle build -->
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>Swagger UI</title>
+  <link rel="stylesheet" type="text/css" href="./swagger-ui.css" >
```

Comment:
```suggestion
  <link rel="stylesheet" type="text/css" href="./swagger-ui.css" >
  <link rel="icon" type="image/png" href="./favicon-32x32.png" sizes="32x32" />
  <link rel="icon" type="image/png" href="./favicon-16x16.png" sizes="16x16" />
```
Not strictly required, but having a favicon would be nice (it is supplied with the swagger files go module).

Reply:
nice idea - added it